nixos/modules/services/networking/keepalived.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.keepalived;
+  keepalivedConf = pkgs.writeText "keepalived.conf" ''
+    ${cfg.extraConfig}
+  '';
+  pidFile = "/run/keepalived.pid";
+
+in
+{
+
+  options = {
+    services.keepalived = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Keepalived.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        description = ''
+          Extra lines to be added verbatim to the configuration file.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.keepalived = {
+      description = "Keepalive Daemon (LVS and VRRP)";
+      after = [ "network.target" "syslog.target" ];
+      wants = [ "network.target" ];
```

Comment:
Explicit dependency on `network.target` is probably redundant, it'll be part of normal bootup anyway. Also, `network.target` is very weakly defined. All you can be sure of is that your service will be stopped *before* network on shutdown ... If you intended to require network connectivity or something, you'd depend on and order after `network-online.target`, that at least is semantically more correct.